### PR TITLE
Typescript & Node.js improvements 

### PR DESCRIPTION
**Pull request by B-Stefan**
Add Typescript ds files 
Add some checks to init the module without error in node.js (I need this for a React-Component )

### CODE REVIEW

**Comment by timdream** [comment]
Thank you for your pull request!  I've just reviewed the code [without white space changes](https://github.com/timdream/google-oauth2-web-client/pull/2?w=1) and I generally agree with all the fixes.

I however disagree with white space changes -- I use 2 spaces consistently through out the code.

Your patch also generates [jshint](https://travis-ci.org/timdream/google-oauth2-web-client/builds/61046475) error preventing me from making sure it passes in the browser test cases.

Please fixes all errors reported by Travis-CI and remove the formatting change from this pull request, and I will accept this. You can run the tests locally with `npm test`. You can force push this branch to re-order and sort/squash your commits.

I have no experience in TypeScript, but you might want to prevent me from breaking your work by adding another test suite that loads and run the script into the environment you are using... as long as it is setup to run on Travis-CI I will accept them too.

Thank you again!

File: src/google-oauth2.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    public onlogin:()=>void;
+    getAccessToken();
+    login(approvalPrompt?:boolean, immediate?:boolean);
+    logout();
+    destory();
+
+}
+declare module "google-oauth2-web-client"{
+    export = GoogleOAuth2
+}
```
Comment (timdream) [inline src/google-oauth2.d.ts]: Need a newline at the end of file to keep git happy.

File: src/google-oauth2.d.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/**
+ * Definition file for the node.js module
+ *
+ */
+interface GoogleOAuth2Options{
```
Comment (timdream) [inline src/google-oauth2.d.ts]: Not entire sure how `ts` files work but it could look nicer if there is a space in front of the opening `{`.

**Comment by k1ng440** [comment]
do you mind if i create a pull request based on this?

**Comment by timdream** [comment]
> do you mind if i create a pull request based on this?

@k1ng440 Not at all, go right ahead and fix it!

**Comment by timdream** [comment]
Merged in #6.

**Pull request closed without merging**